TVR-14652 Improve Agency Proposal documentation (#189)
SAP-docs/preview.developer.concur.com: src/api-reference/request/v4.endpoints.schemas.markdown

`agencyProposalType`|`string`|-|**Required** The agency proposal type value. Supported value: `API`.
`importDate`|`timestamp`|[RFC 3339] Format: `YYYY-MM-DDThh:mm:ss.SSS'Z`|**Read only** The date of the import.
`proposalUuid`|`string`|-|**Read only** The unique identifier of one proposal for a Request.

> `providerMessageId`|`string`|-|The unique identifier of the client’s batch of proposal. All proposals of a given batch should have the same value. Posting a set of proposals using a new different value will remove all proposals previously posted and start a new batch of proposals replacing the previous batch. 

Reviewer: ```suggestion
`providerMessageId`|`string`|-|The unique identifier of the client’s batch of proposal. All proposals of a given batch should have the same value. Posting a set of proposals with a different value will remove all proposals previously posted and start a new batch of proposals, replacing the previous batch.
```

Author: Done

`proposalBatchSize`|`integer`|-|**Required** Size of the client’s batch of proposal. <br>Note: If this value is greater than 1, the Request will not be sent back to the traveler as long as the expected number of proposals (respective proposalOrder) are not integrated into the Request.

> `proposalOrder`|`integer`|-|**Required** Sequence order of the proposal for the request. Up to 3 proposals can be submitted having values 1, 2, 3. 

Reviewer: ```suggestion
`proposalOrder`|`integer`|-|**Required** Sequence order of the proposal for the request. Up to 3 proposals can be submitted with values `1`, `2` and `3`. 
```

Author: Done

`booked`|`boolean`|-|**Required** `true` if this trip is (or has to be) handled by a travel agency.
`approvalLimitDate`|[RFC 3339] `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'`|Limit approval date of the proposal. Past this date, the proposal will not be viable any more. <br>Note: Offset MUST be in form YYYY-MM-DDThh:mm:ss.
`status`|`string`|-|**Required** Supported values: `PROPOSAL`,`CONFIRMED`,`TICKETISSUED`.
`proposal`|`boolean`|-|**Read only** `true` if this “proposal” is still in the `PROPOSAL` status; `false` if the "proposal" status is `CONFIRMED` (tickets issued).
`comments`|`string`|-|Free text.
`itineraryLocator`|`string`|-|The unique identifier for ConcurTravel itinerary. Not provided if this is not a Concur Travel itinerary. 
`selected`|`boolean`|-|**Read only** Value becomes `true` when that proposal is selected by the user.

> `totalPostedAmount`|`object`|[`Amount`](#schema-amount)|The total amount for that proposal. Sum of the agency proposal segments amounts, with currency conversion in case they have prices in different currencies.

Reviewer: ```suggestion
`totalPostedAmount`|`object`|[`Amount`](#schema-amount)|The total amount for that proposal. Sum of the Agency Proposal segments amounts, with currency conversion in case they have prices in different currencies.
```

Name|Type|Format|Description
---|---|---|---
`proposalSegmentUuid`|`string`|-|**Read only** The unique identifier of one segment in a proposal for a request.

> `pnr`|`string`|-|**Required** The personal number of reservation. E.g. `OYY5T`

Reviewer: ```suggestion
`pnr`|`string`|-|**Required** The personal number of reservation (e.g. `OYY5T`).
```

Author: done

`classListItemId`|`object`|[`ResourceLink`](#schema-resourcelink)|The unique identifier of the class of service list item for the segment (will populate the "Class" list field of the segment form - if displayed).

For a given segment along with the common fields above should be included the specific fields for the segment type considered. 
If `proposalSegmentType` is `AIR` then include fields of [`Agency proposal air segment`](#schema-agencyproposalairsegment)
If `proposalSegmentType` is `RAIL` then include fields of [`Agency proposal rail segment`](#schema-agencyproposalrailsegment)
If `proposalSegmentType` is `CAR` then include fields of [`Agency proposal car segment`](#schema-agencyproposalcarsegment)
If `proposalSegmentType` is `HOTEL` then include fields of [`Agency proposal hotel segment`](#schema-agencyproposalhotelsegment)
If `proposalSegmentType` is `MISC` then include fields of [`Agency proposal misc segment`](#schema-agencyproposalmiscsegment)

### <a name="schema-agencyproposalairsegment"></a>Agency proposal air segment

Name|Type|Format|Description
---|---|---|---

> `classOfService`|`string`|-|**Required** The class of the air segment. E.g. `Business`
> `duration`|`integer`|-|**Required** The duration of the booked flight in minutes. Must be greater than 0. E.g. `120` for a two hours flight.
> `flightNumber`|`string`|-|**Required** The flight number for the booking. E.g. `AF029`
> `seatNumber`|`string`|-|The number of the seat. E.g. `24D`

Reviewer: ```suggestion
`classOfService`|`string`|-|**Required** The class of the air segment (e.g. `Business`).
`duration`|`integer`|-|**Required** The duration of the booked flight in minutes. Must be greater than 0 (e.g. `120` for a two hours flight).
`flightNumber`|`string`|-|**Required** The flight number for the booking (e.g. `AF029`).
`seatNumber`|`string`|-|The number of the seat (e.g. `24D`).
```

Author: done

`aircraft`|`string`|-|The code of the aircraft type.

> `departureTerminal`|`string`|-|The departure terminal for the air segment leg. E.g. `2B`

Reviewer: ```suggestion
`departureTerminal`|`string`|-|The departure terminal for the air segment leg (e.g. `2B`).
```

`arrivalTerminal`|`string`|-|The arrival terminal for the air segment leg.
`startLocation`|`object`|[`Location`](#schema-location)|**Required** The start location of the air segment leg (Air location schema).
`endLocation`|`object`|[`Location`](#schema-location)|**Required** The end location of the air segment leg (Air location schema).

### <a name="schema-agencyproposalrailsegment"></a>Agency proposal rail segment

Name|Type|Format|Description
---|---|---|---

> `classOfService`|`string`|-|**Required** The class of the segment leg. E.g. `Economy`
> `duration`|`integer`|-|**Required** The duration of the booked train in minutes. Must be greater than 0. Must be greater than 0. E.g. `120` for a two hours ride.
> `trainNumber`|`string`|-|**Required** The train number for the booking. E.g. `1578`
> `wagonNumber`|`string`|-|The number of the wagon. E.g. `8`
> `seatNumber`|`string`|-|The number of the seat. E.g. `24D`

Reviewer: ```suggestion
`classOfService`|`string`|-|**Required** The class of the segment leg (e.g. `Economy`).
`duration`|`integer`|-|**Required** The duration of the booked train in minutes. Must be greater than 0 (e.g. `120` for a two hours ride).
`trainNumber`|`string`|-|**Required** The train number for the booking (e.g. `1578`).
`wagonNumber`|`string`|-|The number of the wagon (e.g. `8`).
`seatNumber`|`string`|-|The number of the seat (e.g. `24D`).
```

`startRailStation`|`string`|-|The code for the departure station of the ride. See `locationCode` in [`Location`](#schema-location).
`startRailStationName`|`string`|-|The name of the departure station of the ride.
`startLocation`|`object`|[`Location`](#schema-location)|**Required** The start location of the rail segment leg (Rail location schema).
`endRailStation`|`string`|-|The code for the arrival station of the ride. See `locationCode` in [`Location`](#schema-location).
`endRailStationName`|`string`|-|The name of the arrival station of the ride.
`endLocation`|`object`|[`Location`](#schema-location)|**Required** The end location of the rail segment leg (Rail location schema).

## <a name="schema-agencyproposalcarsegment"></a>Agency proposal car segment

Name|Type|Format|Description
---|---|---|---
`airConditioning`|`boolean`|-|Indicates if car has an air conditioner.

> `carEquipment`|`string`|-|Any special equipment required by the renter. E.g. `GPS`
> `classOfService`|`string`|-|The class of the segment leg. E.g. `Executive`

Reviewer: ```suggestion
`carEquipment`|`string`|-|Any special equipment required by the renter (e.g. `GPS`).
`classOfService`|`string`|-|The class of the segment leg (e.g. `Executive`).
```

Author: done

Name|Type|Format|Description
---|---|---|---

> `name`|`string`|-|**Required** The name for the booking. E.g. `Ferry`, `Bus`.

Author: done
